### PR TITLE
add email to google sheet on success pay

### PR DESCRIPTION
**Pull request by wraeth-eth**
Add a row to google sheet for successful payment with an email

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #226** <a href="https://app.graphite.dev/github/pr/peeldao/juicecrowd/226?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #227** <a href="https://app.graphite.dev/github/pr/peeldao/juicecrowd/227?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/peeldao/juicecrowd/227?utm_source=stack-comment).

File: src/app/api/pay/success/route.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import { addRowToGoogleSheet } from '@/lib/backend/google-sheets/addRowToGoogleSheet'
+import { JuiceCrowdError } from '@/lib/error'
+import { etherscanUrlForTx } from '@/lib/etherscan'
+import { publicClient } from '@/lib/viem/publicClient'
+import { NextResponse } from 'next/server'
+import { getAddress } from 'viem'
+import { z } from 'zod'
+
+const TTL = 60n // 1 minute
+const GOOGLE_SPREADSHEET_ID = '1_OpoitLq1pLQIqsdDGh6SbRrdtmABXdIXuJD6HMAVCo'
```
Comment (tomquirk) [inline src/app/api/pay/success/route.ts:12]: @wraeth-eth should we env var? to test on goerl

File: src/app/api/pay/success/route.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    hash: data.transactionHash as `0x${string}`,
+  })
+  if (tx.blockNumber === null) {
+    throw new JuiceCrowdError('Transaction not mined', 400)
+  }
+  if (getAddress(tx.from) !== getAddress(data.walletAddress)) {
+    throw new JuiceCrowdError('Mismatched wallet transaction address', 400)
+  }
+
+  const block = await publicClient.getBlock({ blockHash: tx.blockHash })
```
Comment (tomquirk) [inline src/app/api/pay/success/route.ts:36]: das nice 👏

**Review by tomquirk — APPROVED** [review]

**Comment by wraeth-eth** [comment]
### Merge activity

* **Nov 21, 3:55 PM**: @wraeth-eth started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/peeldao/juicecrowd/227).

**Pull request merged**